apps/wizard/app_pages/insight_search.py

```diff
@@ -0,0 +1,316 @@
+            break
+    return url_img_desktop, url_img_mobile
+
+
+def extract_video_urls_from_raw_data_insight(content) -> str | None:
+    url_video = None
+
+    for element in content.get("body", []):
+        if "type" in element and element["type"] == "video":
+            if "url" in element:
+                url_video = element["url"]
+    return url_video
+
+
+@st.cache_data(show_spinner=False)
+def get_data_insights() -> list[Dict[str, Any]]:
+    with st.spinner("Loading data insights..."):
+        # Get the raw data insights from the database.
+        df = get_raw_data_insights()
+
+        # Parse data insights and construct a convenient dictionary.
+        insights = []
+        for _, di in df.iterrows():
+            content = json.loads(di["content"])
+
+            # Get multimedia urls
+            url_img_desktop, url_img_mobile = extract_image_urls_from_raw_data_insight(content)
+            url_vid = extract_video_urls_from_raw_data_insight(content)
+
+            # Get markdown
+            markdown = di["markdown"]
+            pattern = r"<(Video|Image|Chart)\b[^>]*\/>"
+            if markdown is not None:
+                markdown = re.sub(pattern, "", markdown)
+            else:
+                markdown = extract_text_from_raw_data_insight(content)
+
+            # Build DI dictionary
+            di_dict = {
+                "title": content["title"],
+                "raw_text": extract_text_from_raw_data_insight(content),
+                "authors": content["authors"],
+                "url_img_desktop": url_img_desktop,
+                "url_img_mobile": url_img_mobile,
+                "url_vid": url_vid,
+                "slug": di["slug"],
+                "public": bool(di["published"]),
+                "date_published": di["publishedAt"],
+                "markdown": markdown,
+            }
+
+            if di_dict["public"]:
+                di_dict["url"] = f"https://ourworldindata.org/data-insights/{di_dict['slug']}"
+
+            insights.append(di_dict)
+
+    return insights
+
+
+# @st.cache_data
+# def get_insights_embeddings(insights: list[Dict[str, Any]]) -> list:
+#     # Combine the title and body of each insight into a single string.
+#     insights_texts = [insight["title"] + " " + insight["body"] for insight in insights]
+#     embeddings = [model.encode(doc, convert_to_tensor=True) for doc in tqdm(insights_texts)]
+
+#     return embeddings
+
+
+def _encode_text(text):
+    return MODEL.encode(text, convert_to_tensor=True)
+
+
+@st.cache_data(show_spinner=False)
+def get_insights_embeddings(insights: list[Dict[str, Any]]) -> list:
+    with st.spinner("Generating embeddings..."):
+        # Combine the title, body and authors of each insight into a single string.
+        insights_texts = [
+            insight["title"] + " " + insight["raw_text"] + " " + " ".join(insight["authors"]) for insight in insights
```

Comment:
i think the noise is very minimal, and barely affects the algorithm.

I'm in favour of removing it if we add an author-filter

Reply:
I have just tried searching by author and it doesn't work very well. For certain auhors like Hannah, it does find many DIs, but if you search for either "Pablo Arriagada" or "Pablo Rosado" the first DI is about drug cartels (which is just because of the unfortunate effect of having "Pablo" as part of the embedding 🙃), whose author is actually Bastian. So, if we want to be able to search by author, a dedicated author filter is probably a much better approach.

Reply:
(i) As I said, I agree, filter is better. Once we have one we can drop it.
(ii) Not perfect for all authors, works for some.

Overall, I think is just fine to keep. It adds little (if any) noise, and it works for some authors. Once we have a filter, let's get rid of it.